File: Palindrome_Number/palindrome_number.cpp
#include <iostream>
#include <string>
#include <cmath> // for floor function
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        string str_x = to_string(x);
        bool ans = true;
        for (size_t i = 0; i < floor(str_x.size()/2) ; i++)
        {
            if (str_x[i] == str_x[str_x.size()-i-1])
            {
                ans = ans && true;
            }
            else
            {
                ans = ans && false;
            }
        }
        return ans;
    }
};
int main(){
    Solution s;
    int x = 121;
    bool result = s.isPalindrome(x);
    if (result) {
        cout << x << " is a palindrome." << endl;
    } else {
        cout << x << " is not a palindrome." << endl;
    }

    return 0;
}

File: Palindrome_Number/palindrome_number.py

class Solution:
    def isPalindrome(self, x: int) -> bool:
        x_str = str(x)
        ans = True
        for i in range(len(x_str)//2):
            if x_str[i] == x_str[-i-1]:
                ans = ans and True
            else:
                ans = ans and False

        return ans



# Example usage
if __name__ == "__main__":
    x_int = 121
    solution = Solution()
    result = solution.isPalindrome(x_int)
    print(result)  # Output: [0, 1]